parmed/formats/sdf.py

```diff
@@ -0,0 +1,38 @@
+"""
+This module contains parsers for sdf file format 
+extension described at https://www.cas.org/content/chemical-suppliers/example-sdf
+"""
+from __future__ import print_function, division, absolute_import
+import linecache
+
+from parmed.formats.registry import FileFormatType
+from parmed.utils.six import add_metaclass
+from parmed import rdkit
+
+@add_metaclass(FileFormatType)
+class SDFFile(object):
+    """ Class to read SDF file """
+
+    @staticmethod
+    def id_format(filename):
+        """ Identify the file as a SDF file format or not
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file to test whether or not it is a sdf file
+
+        Returns
+        -------
+        is_fmt : bool
+            True if it is a sdf file, False otherwise
+        """
+        line = linecache.getline(filename, lineno=4)
+        try:
+            return line.split()[-1] in ['V2000', 'V3000']
```

Comment:
Going based on the file format spec at https://en.wikipedia.org/wiki/Chemical_table_file#Molfile, I would make one small change.  That line says that the first two numbers are the number of atoms and number of bonds, and gives no meaning to the rest of the numbers.  What this tells me is that that field *must* have at least 3 words.  So I would change this to the following:

```python
words = linecache.getline(filename, lineno=4).split()
return len(words) >= 3 and words[-1] in ('V2000', 'V3000')
```

Easy peasy :)